### PR TITLE
Request to Add Deploying XenForo to Compute Engine Tutorial

### PR DESCRIPTION
**Pull request by michaelawyu**
- [x] Tech writing / styleguide review - @ryanmcginnis  
- [x] Formatting review for publishing - @jmdobry

### CODE REVIEW

**Comment by ryanmcginnis** [comment]
Apologies for the delay. I will review this tutorial presently.

File: tutorials/deploy-xenforo-to-compute-engine/index.md [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+	sudo apt-get update
+	```
+
+2. Install Apache server and its module for PHP support:
+
+	```bash
+	sudo apt-get install apache2 apache2-mod-php7.0 -y
+	```
+
+	After the installation is completed, open your browser and visit http://`[SERVER_EXTERNAL_IP]` (note that it is a HTTP link instead of a HTTPS link). `[SERVER_EXTERNAL_IP]` can be found on the [VM Instances](https://console.cloud.google.com/compute/instances) page, in the row of the instance that you connect to. You should see a page titled “Apache Debian Default Page” in your browser.
```
Comment (ryanmcginnis) [inline tutorials/deploy-xenforo-to-compute-engine/index.md]: "After the installation is complete,"

Comment (ryanmcginnis) [inline tutorials/deploy-xenforo-to-compute-engine/index.md]: "[VM Instances](https://console.cloud.google.com/compute/instances) menu in the Cloud Platform Console"

File: tutorials/deploy-xenforo-to-compute-engine/index.md [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+	sudo apt-get install mariadb-server -y
+	```
+
+4. Set up MariaDB:
+
+	```bash
+	sudo mysql_secure_installation
+	```
+
+	The script prepares MariaDB for production use. For security reasons you should follow the prompts on screen, set up a root password, remove all anonymous users, disable remote login for root user, remove test database and reload the privilege tables.
```
Comment (ryanmcginnis) [inline tutorials/deploy-xenforo-to-compute-engine/index.md]: "For security reasons, you should follow the on-screen prompts, set up a root password, remove all anonymous users, disable remote login for the root user, remove test database, and reload the privilege tables."

File: tutorials/deploy-xenforo-to-compute-engine/index.md [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+
+	Then enter the following commands in MariaDB monitor:
+
+	```sql
+	CREATE DATABASE [DATABASE_NAME];
+	CREATE USER ‘[DATABASE_USER]’@’localhost’ IDENTIFIED BY ‘[DATABASE_PASSWORD]’;
+	GRANT ALL PRIVILEGES ON *.* TO ‘[DATABASE_USER]’@’localhost’;
+	```
+
+	Commands above create a new database and a new database user with full privileges in the system. Replace `[DATABASE_NAME]`, `[DATABASE_USER]` and `[DATABASE_PASSWORD]` with values of your own. You may want to write down those values as you need to use them later when installing XenForo.
```
Comment (ryanmcginnis) [inline tutorials/deploy-xenforo-to-compute-engine/index.md]: "There commands create a new database and a new database user with full privileges in the system. Replace `[DATABASE_NAME]`, `[DATABASE_USER]` and `[DATABASE_PASSWORD]` with your own values. You might want to write down these values, as you need them to install XenForo."

File: tutorials/deploy-xenforo-to-compute-engine/index.md [hunk 4]
```diff
@@ -0,0 +1,267 @@
...
+	sudo apt-get install php7.0 php7.0-mysql php7.0-gd php7.0-xml -y
+	```
+
+7. Install additional tools and packages:
+
+	```bash
+	sudo apt-get install unzip postfix libsasl2-modules
+	```
+
+	Not all of the tools and packages listed above are required XenForo; however you may find them useful when installing and configuring XenForo.
```
Comment (ryanmcginnis) [inline tutorials/deploy-xenforo-to-compute-engine/index.md]: "Although not all of the tools and packages listed above are required for XenForo, you might find them useful during installation and configuration."

File: tutorials/deploy-xenforo-to-compute-engine/index.md [hunk 5]
```diff
@@ -0,0 +1,267 @@
...
+
+	During the installation of postfix a configuration screen will come up. Use TAB key to move the cursor, UP and DOWN keys to change the choice and Enter to confirm. For this tutorial we will use the `Local Only` type. You can leave other settings unchanged.
+
+	Now the instance is ready for running XenForo.
```
Comment (ryanmcginnis) [inline tutorials/deploy-xenforo-to-compute-engine/index.md]: "Now the instance is ready to run XenForo. "

File: tutorials/deploy-xenforo-to-compute-engine/index.md [hunk 6]
```diff
@@ -0,0 +1,267 @@
...
+
+7. Install additional tools and packages:
+
+	```bash
+	sudo apt-get install unzip postfix libsasl2-modules
+	```
+
+	Not all of the tools and packages listed above are required XenForo; however you may find them useful when installing and configuring XenForo.
+
+	During the installation of postfix a configuration screen will come up. Use TAB key to move the cursor, UP and DOWN keys to change the choice and Enter to confirm. For this tutorial we will use the `Local Only` type. You can leave other settings unchanged.
```
Comment (ryanmcginnis) [inline tutorials/deploy-xenforo-to-compute-engine/index.md]: "During the installation of postfix, a configuration screen is presented. Use the TAB key to move the cursor, UP and DOWN keys to change the choice, and Enter to confirm. For this tutorial, use the `Local Only` type and leave other settings unchanged."

File: tutorials/deploy-xenforo-to-compute-engine/index.md [hunk 7]
```diff
@@ -0,0 +1,267 @@
...
+
+	Now the instance is ready for running XenForo. 
+
+## Uploading and Configuring XenForo
+
+1. Copy the XenForo installation file to your Compute Engine instance:
+
+	* Open [Cloud Storage Browser](https://console.cloud.google.com/storage/browser) page and click Create **Bucket**. Enter a name for the new bucket and choose a storage class for the bucket in the new page. For this tutorial you should create a regional bucket in the same location as your Compute Engine instance.
```
Comment (ryanmcginnis) [inline tutorials/deploy-xenforo-to-compute-engine/index.md]: "Visit the [Cloud Storage Browser](https://console.cloud.google.com/storage/browser) menu in the Cloud Platform Console"

Comment (ryanmcginnis) [inline tutorials/deploy-xenforo-to-compute-engine/index.md]: "on the new page"

Comment (ryanmcginnis) [inline tutorials/deploy-xenforo-to-compute-engine/index.md]: "For this tutorial, create a regional bucket in the same location as your Compute Engine instance."

File: tutorials/deploy-xenforo-to-compute-engine/index.md [hunk 8]
```diff
@@ -0,0 +1,267 @@
...
+	* Click **Upload Files** and choose your XenForo installation file. It should be a .zip file downloaded from XenForo website. If you have Cloud SDK installed on your system, you can also [use the gsutil tool to upload the file](https://cloud.google.com/storage/docs/gsutil/commands/cp).
```
Comment (ryanmcginnis) [inline tutorials/deploy-xenforo-to-compute-engine/index.md]: [use the `gsutil` tool to... ]

File: tutorials/deploy-xenforo-to-compute-engine/index.md [hunk 9]
```diff
@@ -0,0 +1,267 @@
...
+
+	* Change directory to Apache web root directory, grant file access permission XenForo requires and remove the default homepage (`index.html`):
+
+	    ```bash
+	    cd /var/www/html
+	    sudo chmod -R 777 *
+	    sudo rm index.html
+	    ```
+
+	* Open your browser and visit https://`[SERVER_EXTERNAL_IP]`. As shown in previous steps, you can find the external IP of your Compute Engine instance on the [VM Instances](https://console.cloud.google.com/compute/instances) page, in the row of the instance that you connect to. Follow the prompts on screen to configure your XenForo installation. During installation XenForo will ask you for credentials to connect to your MySQL server. Your MySQL Server and MySQL Port are localhost and 3306 respectively. Your MySQL User Name, MySQL Password and MySQL Database Name are the values you entered previously when installing MariaDB.
```
Comment (ryanmcginnis) [inline tutorials/deploy-xenforo-to-compute-engine/index.md]: Wrap in blackticks: `https://[SERVER_EXTERNAL_IP]`

File: tutorials/deploy-xenforo-to-compute-engine/index.md (hunk 1 above)
Comment (ryanmcginnis) [inline tutorials/deploy-xenforo-to-compute-engine/index.md]: Wrap in blackticks: `https://[SERVER_EXTERNAL_IP]`

File: tutorials/deploy-xenforo-to-compute-engine/index.md [hunk 10]
```diff
@@ -0,0 +1,267 @@
...
+
+Your XenForo website is now up and running.
+
+## Setting Up a Mailing Service
+
+XenForo by default uses a mail transfer agent (MTA), such as postfix, to send emails. The agent then utilizes an outgoing connection to common SMTP ports (25, 587, etc.) to transfer information (mails). However, many Cloud service providers, including Google Cloud Platform, block such connections for security reasons; additionally, emails sent this way are often considered spams by many email service providers, and will be dropped before getting into the inboxes of recipients. For those reasons, you should use a third-party service for sending emails from your XenForo website.
```
Comment (ryanmcginnis) [inline tutorials/deploy-xenforo-to-compute-engine/index.md]: "many cloud service providers"

Comment (ryanmcginnis) [inline tutorials/deploy-xenforo-to-compute-engine/index.md]: "considered spam"

Comment (ryanmcginnis) [inline tutorials/deploy-xenforo-to-compute-engine/index.md]: "and are dropped before reaching the inboxes of their recipients"

Comment (ryanmcginnis) [inline tutorials/deploy-xenforo-to-compute-engine/index.md]: "For these reasons, you should use a third-party service to send emails from your XenForo website."

File: tutorials/deploy-xenforo-to-compute-engine/index.md [hunk 11]
```diff
@@ -0,0 +1,267 @@
...
+
+[SendGrid](https://sendgrid.com/), [Mailgun](https://www.mailgun.com/), and [Mailjet](https://www.mailjet.com/) are Compute Engine’s third-party partners for sending emails. For more information on Compute Engine’s partnerships with those services, refer to [Sending Email with SendGrid](https://cloud.google.com/compute/docs/tutorials/sending-mail/using-sendgrid), [Sending Email with Mailgun](https://cloud.google.com/compute/docs/tutorials/sending-mail/using-mailgun) and [Sending Email with Mailjet](https://cloud.google.com/compute/docs/tutorials/sending-mail/using-mailjet). If you plan to use other services, follow instructions on their websites to set up SMTP Relay for your XenForo website.
```
Comment (ryanmcginnis) [inline tutorials/deploy-xenforo-to-compute-engine/index.md]: Change to "these services" and add third comma:

"For more information on Compute Engine’s partnerships with **these** services, refer to [Sending Email with SendGrid](https://cloud.google.com/compute/docs/tutorials/sending-mail/using-sendgrid), [Sending Email with Mailgun](https://cloud.google.com/compute/docs/tutorials/sending-mail/using-mailgun)**,** and [Sending Email with Mailjet](https://cloud.google.com/compute/docs/tutorials/sending-mail/using-mailjet)."

File: tutorials/deploy-xenforo-to-compute-engine/index.md [hunk 12]
```diff
@@ -0,0 +1,267 @@
...
+	cd /etc/postfix
+	```
+
+3. Edit the preferences:
+
+	```bash
+	nano main.cf
+	```
+
+	Use arrow keys (UP, DOWN, LEFT and RIGHT keys) to move the cursor around. Comment out the following lines:
```
Comment (ryanmcginnis) [inline tutorials/deploy-xenforo-to-compute-engine/index.md]: "UP, DOWN, LEFT, and RIGHT keys"

File: tutorials/deploy-xenforo-to-compute-engine/index.md [hunk 13]
```diff
@@ -0,0 +1,267 @@
...
+	| default_transport = error | #default_transpot = error |
+	| relay_transport = error   | #relay_transport = error  |
+
+	Edit the following lines:
+
+	| Before      | After                             |
+	| ------------|-----------------------------------|
+	| relayhost = | relayhost = [RELAY_SERVER]:[PORT] |
+
+	where the `[RELAY_SERVER]` and `[PORT]` are provided by the mailing service. Note that you cannot use common SMTP ports such as 25, 465 or 587. If you are using SendGrid, Mailgun or Mailjet, here are their relay services and ports for Compute Engine instances, as seen in the [Compute Engine documentation](https://cloud.google.com/compute/docs/tutorials/sending-mail/):
```
Comment (ryanmcginnis) [inline tutorials/deploy-xenforo-to-compute-engine/index.md]: Comma: "SendGrid, Mailgun, or Mailjet"

File: tutorials/deploy-xenforo-to-compute-engine/index.md [hunk 14]
```diff
@@ -0,0 +1,267 @@
...
+
+	```
+	[RELAY_SERVER]:[PORT] [SMTP_USERNAME]:[SMTP_PASSWORD]
+	```
+
+	Replace `[RELAY_SERVER]`, `[PORT]`, `[SMTP_USERNAME]` and `[SMTP_PASSWORD]` with the values you used or recorded earlier.
+
+	Press Ctrl + O (Control Key and O Key together) to write the file. Then press Ctrl + X (Control Key and X Key together) to exit.
+
+	Convert the file you just created to a .db file, grant proper access permission to it and remove the old file with
```
Comment (ryanmcginnis) [inline tutorials/deploy-xenforo-to-compute-engine/index.md]: Comma and additional language: "Convert the file you just created to a .db file, grant proper access permission to it, and remove the old file with the following commands:"

File: tutorials/deploy-xenforo-to-compute-engine/index.md [hunk 15]
```diff
@@ -0,0 +1,267 @@
...
+	sudo rm sasl_password
+	```
+
+5. Lastly, restart the postfix service:
+
+	```bash
+	sudo service postfix restart
+	```
+
+	And now your XenForo website can use the third-party service you choose to send emails more securely and efficiently.
```
Comment (ryanmcginnis) [inline tutorials/deploy-xenforo-to-compute-engine/index.md]: "Now your XenForo website can use the third-party service to send emails more securely and efficiently."

File: tutorials/deploy-xenforo-to-compute-engine/index.md [hunk 16]
```diff
@@ -0,0 +1,267 @@
...
+
+## Update Your Domain (Optional)
+
+Anyone can now use the external IP of your Compute Engine instance to access your XenForo website. To assign a domain name to this IP address, follow the steps below:
```
Comment (ryanmcginnis) [inline tutorials/deploy-xenforo-to-compute-engine/index.md]: "To assign a domain name to this IP address, perform the following steps:"

File: tutorials/deploy-xenforo-to-compute-engine/index.md [hunk 17]
```diff
@@ -0,0 +1,267 @@
...
+
+1. Assign a static IP to your Compute Engine instance:
+
+	Currently the external IP you see on the [VM Instances](https://console.cloud.google.com/compute/instances) page changes every time you restart the instance. You need to assign a static IP address to it to make sure that your website is always accessible even if the server restarts. Go to the [External IP addresses](https://console.cloud.google.com/networking/addresses/list) page, and click **Ephemeral** in the row of the Compute Engine instance hosting your XenForo website. In the drop-down menu, choose **Static**. Enter a new name for this static address and click **Reserve**.
```
Comment (ryanmcginnis) [inline tutorials/deploy-xenforo-to-compute-engine/index.md]: "the [VM Instances](https://console.cloud.google.com/compute/instances) menu"

Comment (ryanmcginnis) [inline tutorials/deploy-xenforo-to-compute-engine/index.md]: "Visit the [External IP addresses](https://console.cloud.google.com/networking/addresses/list) menu and click **Ephemeral** in the row of the Compute Engine instance hosting your XenForo website. From the drop-down menu, choose **Static**. Enter a new name for this static address, then click **Reserve**."

File: tutorials/deploy-xenforo-to-compute-engine/index.md [hunk 18]
```diff
@@ -0,0 +1,267 @@
...
+
+	If you have Cloud SDK installed in the system, you can also [use the gcloud command-line to reserve a static IP address](https://cloud.google.com/compute/docs/ip-addresses/reserve-static-external-ip-address).
```
Comment (ryanmcginnis) [inline tutorials/deploy-xenforo-to-compute-engine/index.md]: Blackticks: "[use the `gcloud` command-line to ... ]

File: tutorials/deploy-xenforo-to-compute-engine/index.md [hunk 19]
```diff
@@ -0,0 +1,267 @@
...
+
+2. If you haven’t done so, get a domain name from domain name registrars. Follow their instructions to add the IP address you just reserved to your domain. Your website should be accessible with the domain name in a short while.
+
+## Cleaning Up
+
+After you have finished this tutorial, you can clean up the resources you created on Google Cloud Platform so that you will not be billed for them in the future. To clean up, you can delete the whole project or stop the Compute Engine instance.
+
+### Deleting the Project
+
+Go to the [Manage resources](https://console.cloud.google.com/cloud-resource-manager) page. Select the project you used for this tutorial, and click **Delete**. Note that once the project is deleted, the project ID cannot be reused.
```
Comment (ryanmcginnis) [inline tutorials/deploy-xenforo-to-compute-engine/index.md]: "Visit to the [Manage resources](https://console.cloud.google.com/cloud-resource-manager) menu"

Comment (ryanmcginnis) [inline tutorials/deploy-xenforo-to-compute-engine/index.md]: Remove comma: "Select the project you used for this tutorial and click **Delete**."

File: tutorials/deploy-xenforo-to-compute-engine/index.md [hunk 20]
```diff
@@ -0,0 +1,267 @@
...
+
+If you have Cloud SDK installed in the system, you can also [use the gcloud command-line to delete a project](https://cloud.google.com/sdk/gcloud/reference/projects/delete).
```
Comment (ryanmcginnis) [inline tutorials/deploy-xenforo-to-compute-engine/index.md]: Blackticks: "use the `gcloud` command-line to ..."

File: tutorials/deploy-xenforo-to-compute-engine/index.md [hunk 21]
```diff
@@ -0,0 +1,267 @@
...
+
+### Stopping the Compute Engine Instance
+
+Go the [VM Instances](https://console.cloud.google.com/compute/instances) page. Select the instance you used for this tutorial, and click **Stop**. If you have reserved a static IP address, it charges you at an hourly rate when the address is not used. To release the address, go to [External IP addresses](https://console.cloud.google.com/networking/addresses/list) page, select the address and click **Release Static Address**. You may also want to go to the Cloud Storage [browser](https://console.cloud.google.com/storage/browser) page and delete the bucket you created during the tutorial.
```
Comment (ryanmcginnis) [inline tutorials/deploy-xenforo-to-compute-engine/index.md]: "Visit the [VM Instances](https://console.cloud.google.com/compute/instances) menu. Select the instance you used for this tutorial and click **Stop**. If you have reserved a static IP address, you are charged at an hourly rate when the address is not used. To release the address, go to [External IP addresses](https://console.cloud.google.com/networking/addresses/list) menu, select the address, and click **Release Static Address**. You might also want to go to the Cloud Storage [browser](https://console.cloud.google.com/storage/browser) page and delete the bucket you created during the tutorial"

File: tutorials/deploy-xenforo-to-compute-engine/index.md [hunk 22]
```diff
@@ -0,0 +1,267 @@
...
+	* Grant your instance full access to all Cloud APIs.
+	* Allow HTTP and HTTPS traffic to your instance.
+	* Click Create. It takes a short while for Compute Engine to create and start your instance.
+
+	Depending on the scale of your website, you may need to add more CPU cores, increase the amount of memory available and use a large SSD persistent boot disk instead of the default 10GB standard one. Though for this tutorial it is OK to leave these settings unchanged. It is also possible to upgrade (or downgrade) your instance after deployment.
+
+	For instructions on creating and starting a Compute Engine instance via Cloud SDK or with Compute Engine API, refer to the [Creating and Starting an Instance page](https://cloud.google.com/compute/docs/instances/create-start-instance) in Compute Engine documentation.
+
+5. Download your copy of XenForo from XenForo’s website. For this instruction you need to use XenForo 1.5.
```
[the remark this answers is not in the document]
Comment (michaelawyu) [reply tutorials/deploy-xenforo-to-compute-engine/index.md]: XenForo is a commercial software; customers need to log in with their accounts to get their download links. Should I change it to "Sign in with your account and download your copy of XenForo from the page"?

[the remark this answers is not in the document]
Comment (ryanmcginnis) [reply tutorials/deploy-xenforo-to-compute-engine/index.md]: Yes.

**Comment by michaelawyu** [comment]
Thanks so much, @ryanmcginnis !

**Pull request merged**